src/LockManager.php

```diff
@@ -209,6 +221,10 @@ public function unlock(Lock $lock)
 			$this->debug('> sent unlock request to '. $this->servers[$i][0] .':'. $this->servers[$i][1]);
 		}
 		
+		if (count($this->sockets) < $this->quorum) {
```

Comment:
If we can't reach enough servers to form a quorum, any locks held are implicitly released.  I'd say this isn't necessary here.